nixos/dovecot: do not include empty sieve_extensions and sieve_global_extensions

exi opened the pull request:
First try [https://github.com/NixOS/nixpkgs/pull/281915](https://github.com/NixOS/nixpkgs/pull/281915) failed because everyone was missing missing newlines...

This PR squashes all the previous commits and adds proper newlines.

@2xsaiko @dotlambda @h7x4 terribly sorry for the noise

This time I properly built my local config from this version and the config comes out as this:
```
plugin {
  sieve_plugins = sieve_imapsieve sieve_extprograms
sieve_extensions = +notify +imapflags +vnd.dovecot.filter
sieve_global_extensions = +vnd.dovecot.environment +vnd.dovecot.pipe
```

Looks alright and works on my server

dotlambda reviewed (changes requested):
Use https://github.com/NixOS/nixpkgs/pull/281915#discussion_r1462335626 and fix the commit message.

exi commented:
> Use [#281915 (comment)](https://github.com/NixOS/nixpkgs/pull/281915#discussion_r1462335626) and fix the commit message.

@dotlambda Thanks for the review. I did the changes and force-pushed, which seems to break the "review conversations" features in github. I just re-requested a review, I hope that was the right thing to do.

exi commented:
@dotlambda Github does not allow me to address your change. After the force-push to correct the commit message of the commit/file you commented on, there is no way to get back to the change you commented on to mark it addressed :(

dblsaiko reviewed (approved):
Tested, works for me.

dotlambda commented:
I think it makes more sense to use `nullOr listOf` because maybe some people want to explicitly disable all extensions.

exi commented:
> I think it makes more sense to use `nullOr listOf` because maybe some people want to explicitly disable all extensions.

@dotlambda @2xsaiko I changed the module type to `nullOr listOf`, tested it and it works for me (tm). Also tested the case of manually setting the list to `[ ]` to see that I get an empty entry at the end to override the defaults.

exi committed:
nixos/dovecot: Do not include sieve_extensions and sieve_global_extensions if they are the default value

Setting them to empty string will disable the default behaviour, leading to missing extensions.
Also set the defaults to null to make them explicitly overridable with
empty lists.

exi commented:
I will close this PR for now. I need to rework it because the underlying file has changes substantially in the last 2 days on master.

dotlambda commented:
You should convert it to a draft instead.
